### PR TITLE
Record notes on hashing artifacts in a monorepo

### PR DESCRIPTION
**Pull request by jtmcdole**
aka "dude, where's my engine?"

### CODE REVIEW

**Comment by jtmcdole** [comment]
Adding yegor since he was interesting in A/B testing in flutter/flutter#156027

File: docs/monorepo/engine_binary_hashing.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+```bash
+git ls-tree -R $(git merge-base HEAD master) engine DEPS | git hash-object --stdin
+```
+
+## Considerations and Future Refinements
+
+Currently, `git ls-tree` incorporates file names and permissions into the hash calculation. Consequently, renaming or changing permissions of a file will necessitate rebuilding the engine. While acceptable initially, this behavior might require adjustments in the future.
+
+If we want to focus solely on file content and size, we could use `git ls-tree -r --object-only engine DEPS | sort | git hash-object --stdin`. However, this relies on consistent sorting across operating systems, which might introduce complexities.
```
Comment (zanderso) [inline docs/monorepo/engine_binary_hashing.md:71]: Sorry, just skimming so maybe you address this, but I'm noticing here that the output of `ls-tree` is sorted before handing to `git hash-object`, but *isn't* sorted below in the command down on line 93.

Comment (jtmcdole) [reply docs/monorepo/engine_binary_hashing.md:71]: This section is talking about tuning the output of ls-tree to only output the Sha1 sum and then sorting it. That should make the hash resilient to files moving and being renamed.

However, this use case is fairly specialized and potentially rare. It's probably not worth the complication right now.

Comment (jtmcdole) [reply docs/monorepo/engine_binary_hashing.md:71]: Line 93 has the internal sort for git trees.

Comment (zanderso) [reply docs/monorepo/engine_binary_hashing.md:71]: Got it. Maybe `Considerations and Future Refinements` should come after `Recommended Formula and Implementation`.

Comment (jtmcdole) [reply docs/monorepo/engine_binary_hashing.md:71]: Sure thing

**Review by zanderso — APPROVED** [review]

**Pull request merged**